flow-nodes/api-builder-plugin-format-date/src/actions.js

```diff
@@ -21,25 +21,29 @@ const moment = require('moment');
  *	 does not define "next", the first defined output).
  */
 function formatDate(params) {
-	const { date, format } = params;
+  const { date, format, offset } = params;
 
-	if (!date) {
-		throw new Error('Missing required parameter: date');
-	}
-	if (!format) {
-		throw new Error('Missing required parameter: format');
-	}
+  // Defaults to now
+  const momentObj = moment(date);
 
-	// There is no sanity check on the formats in Moment.js, so an invalid format could be
-	// specified. As this is a wrapper, I won't be introducing additional format validation.
-	const formattedDate = moment(date).format(format);
+  if (!momentObj.isValid()) {
+ 		throw new Error(`Invalid date: ${date}`);
+ 	}
 
-	// Moment doesn't throw an error if the passed in date is invalid it just returns 'Invalid date'
-	if (formattedDate === 'Invalid date') {
-		throw new Error(`Invalid date: '${date}`);
-	}
-	return formattedDate;
-}
+  if (offset) {
+    const pattern = new RegExp('^[+-][0-1][0-9]:[0-5][0-9]$');
+    const isValidOffset = pattern.test(offset);
+    if (isValidOffset === true) {
```

Comment:
`if (pattern.test(offset)) {`

you'll rarely want to test booleans with `=== true` or `=== false`